src/Maker/AdminMaker.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Maker;
+
+use Sonata\AdminBundle\Command\Validators;
+use Sonata\AdminBundle\Manipulator\ServicesManipulator;
+use Sonata\AdminBundle\Model\ModelManagerInterface;
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Bundle\MakerBundle\Maker\AbstractMaker;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\DependencyInjection\Container;
+
+/**
+ * @author Gaurav Singh Faudjdar <faujdar@gmail.com>
+ */
+class AdminMaker extends AbstractMaker
+{
+    /**
+     * @var string
+     */
+    private $projectDirectory;
+
+    /**
+     * @var ModelManagerInterface[]
+     */
+    private $availableModelManagers;
+
+    public function __construct($projectDirectory, array $modelManagers = [])
+    {
+        $this->projectDirectory = $projectDirectory;
+        $this->availableModelManagers = $modelManagers;
+    }
+
+    /**
+     * Return the command name for your maker (e.g. make:report).
+     *
+     * @return string
+     */
+    public static function getCommandName(): string
+    {
+        return 'make:sonata:admin';
+    }
+
+    /**
+     * Configure the command: set description, input arguments, options, etc.
+     *
+     * By default, all arguments will be asked interactively. If you want
+     * to avoid that, use the $inputConfig->setArgumentAsNonInteractive() method.
+     *
+     * @param Command            $command
+     * @param InputConfiguration $inputConfig
+     */
+    public function configureCommand(Command $command, InputConfiguration $inputConfig)
+    {
+        $command
+            ->setDescription('Generates an admin class based on the given model class')
+            ->addArgument('model', InputArgument::REQUIRED, 'The fully qualified model class')
+            ->addOption('admin', 'a', InputOption::VALUE_OPTIONAL, 'The admin class basename')
+            ->addOption('controller', 'c', InputOption::VALUE_OPTIONAL, 'The controller class basename')
+            ->addOption('manager', 'm', InputOption::VALUE_OPTIONAL, 'The model manager type')
+            ->addOption('services', 'y', InputOption::VALUE_OPTIONAL, 'The services YAML file', 'services.yaml')
```

Comment:
why `y` and not `s`?

Reply:
for consistency with the old command

Reply:
we should use `s` here, the command is new, so no need to keep `y` IMO